workflow-templates/sphinx.yml

```diff
@@ -0,0 +1,40 @@
+# Builds the Sphinx documentation in the specified folder.
+# Default values are docs/ for the directory where the Sphinx
+# documentation is located and true to run linkcheck
+# To pass a new folder, set directory_path to the folder path
+# To disable linkcheck, set check_link to any value 
+
+---
+name: sphinx
+
+on: 
+  push:
+  pull_request:
+  schedule:
+    - cron: '0 0 * * 0'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    env: 
+      directory_path:
+      check_link: true # set to false to remove linkcheck
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - name: Sphinx install
+        run: pip install --upgrade sphinx
+      - name: Set directory
+        run: echo "docs_dir=./docs" >> $GITHUB_ENV
+        if: ${{ env.directory_path == '' }}
+      - name: Set directory
+        run: echo "docs_dir="${{ env.directory_path }} >> $GITHUB_ENV
+        if: ${{ env.directory_path != '' }}
```

Comment:
I haven't yet found an example, but I suspect it's possible to shorten this pattern here by mixing `{{ }}` and bash-isms (and possibly even `env:` blocks)

Reply:
I have not found the way yet!!

Reply:
Did you try something like this?
```
$ A=""
$ echo "${A:-foo}"
foo
```
That _might_ be here:
```
run: echo "docs_dir="${${{ env.directory_path }}:-./docs}" >> $GITHUB_ENV
```
though the first `$` may need escaping.

Reply:
I did not try that combination, I will try it out

Reply:
Ok I managed to simplify things
https://github.com/jburel/omero-guide-python/runs/1418905984?check_suite_focus=true
The substitution must happen in 2 steps to avoid a `` bad substitution`` error